cp_core/src/task/transform/common.rs

```diff
@@ -209,4 +261,95 @@ mod tests {
         };
         rt.block_on(event());
     }
+
+    #[test]
+    fn create_root_transform_good_config() {
+        let select_yaml = r#"
+select:
+    key: value
+"#;
+        let select_value: serde_yaml_ng::Value = serde_yaml_ng::from_str(select_yaml).unwrap();
+
+        let join_yaml = r#"
+join:
+    right: BASIC
+    left_on: [col]
+    right_on: [my_col]
+    how: left
+"#;
+        let join_value: serde_yaml_ng::Value = serde_yaml_ng::from_str(join_yaml).unwrap();
+
+        let config = RootTransformConfig {
+            label: "test_label".to_string(),
+            input: StrKeyword::with_value("test_input".to_owned()),
+            output: StrKeyword::with_value("test_output".to_owned()),
+            subtransforms: vec![select_value, join_value],
+        };
+
+        assert!(config.validate().is_empty());
+    }
+
+    #[test]
+    fn create_root_transform_bad_config() {
+        let select_yaml = r#"
+select:
+    key: value
+"#;
+        let select_value: serde_yaml_ng::Value = serde_yaml_ng::from_str(select_yaml).unwrap();
+
+        let invalid_yaml = r#"
+purr_transform:
+    key: value
+"#;
+        let invalid_value: serde_yaml_ng::Value = serde_yaml_ng::from_str(invalid_yaml).unwrap();
+
+        let config = RootTransformConfig {
+            label: "test_label".to_string(),
+            input: StrKeyword::with_value("test_input".to_owned()),
+            output: StrKeyword::with_value("test_output".to_owned()),
+            subtransforms: vec![select_value, invalid_value],
+        };
+
+        assert_eq!(config.validate().len(), 1);
+    }
+
+    #[test]
+    fn valid_root_transform_basic() {
+        let select_yaml_1 = r#"
```

Comment:
what does the `r#` mean? I actually don't know this

Reply:
It's like raw string, but I think you can use `r#`, or `r##` etc as delimiters in case you have a e.g. `"#` as a valid part of your string that you want to represent.